src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -220,9 +240,34 @@ private void RemovePlaceholderScope(BoundValuePlaceholderBase placeholder)
 
         private uint GetPlaceholderScope(BoundValuePlaceholderBase placeholder)
         {
-            Debug.Assert(_placeholderScopes is { });
-            return _placeholderScopes[placeholder];
+            Debug.Assert(_placeholderScopes?.ContainsKey(placeholder) == true);
+
+            return _placeholderScopes?.TryGetValue(placeholder, out var scope) == true
+                ? scope
+                : CallingMethodScope;
+        }
+
+#if DEBUG
+        private bool ContainsPlaceholderScope(BoundValuePlaceholderBase placeholder)
```

Comment:
nit: this method is only used in asserts, consider renaming this method `AssertContainsPlaceholderScope` and making it void-returning and conditional, with assertions instead of return statements for failure cases.

Reply:
I think I'd rather leave as a `bool` returning method where the callers assert.